lib/web_ui/lib/src/ui/window.dart

```diff
@@ -122,6 +122,8 @@ class WindowPadding {
   static const WindowPadding zero =
       WindowPadding._(left: 0.0, top: 0.0, right: 0.0, bottom: 0.0);
 
+  WindowPadding.forKeyboard({this.left, this.top, this.right, this.bottom});
```

Comment:
`WindowPadding` is a public-facing class that is shared with the native engine. Any change in its API should also be made in the native engine.

The approach we have been following is keeping the class in `src/ui/` abstract, and provide an implementation in `src/engine/` which is private to the web engine. See `class Window` and `class Paragraph` for an example of how to do this.

Reply:
I'm not sure how to make abstract for this specific class. It is defined as a concrete class in src/ui, also other classes inside src/ui, uses it. 

Shall I keep it as it is but create another class in the engine such as EngineWindowPadding?

Reply:
The trick is to make the class abstract, but give it a constructor that delegates to `EngineWindowPadding`'s constructor.

```dart
// in src/ui:

abstract class WindowPadding {
  const factory WindowPadding._({
    double left,
    double top,
    double right,
    double bottom,
  }) = engine.WindowPadding;
  
  static const WindowPadding zero =
      WindowPadding._(left: 0.0, top: 0.0, right: 0.0, bottom: 0.0);

  double get left;
  double get top;
  double get right;
  double get bottom;
}

// in src/engine:

class WindowPadding implements ui.WindowPadding {
  const WindowPadding({
    this.left,
    this.top,
    this.right,
    this.bottom,
  });
  
  final double left;
  final double top;
  final double right;
  final double bottom;
  
  @override
  String toString() {
    return 'WindowPadding(left: $left, top: $top, right: $right, bottom: $bottom)';
  }
}
```
Now in your engine code, you can do `WindowPadding(bottom: bottomPadding, left: 0, right: 0, top: 0);`.

Reply:
done! thanks for the suggestion.